src/cloudant/replicator.py

```diff
@@ -77,31 +80,30 @@ def create_replication(self, source_db=None, target_db=None,
         if not data.get('source'):
             if source_db is None:
                 raise CloudantException(
-                    "You must specify either a source_db Database "
-                    "object or a manually composed 'source' string/dict."
+                    'You must specify either a source_db Database '
+                    'object or a manually composed \'source\' string/dict.'
+                )
+            data['source'] = {'url': source_db.database_url}
+            if not source_db.cloudant_account.admin_party:
```

Comment:
I think I would maybe proxy this property with a property on the database object:

```python
@property
 def admin_party(self):
       return self.cloudant_account.admin_party 

```

save a bit of boilerplate